classes/DepositPackage.inc.php

```diff
@@ -598,68 +606,77 @@ public function updateDepositStatus() {
 					'processingState' => $processingState)),
 				SCHEDULED_TASK_MESSAGE_TYPE_NOTICE);
 
+		// Clear previous error messages
 		$this->_deposit->setExportDepositError(null);
+		$this->_deposit->setStagingState($processingState ?: null);
+		// Handle the local state
 		switch ($processingState) {
 			case 'depositedByJournal':
-				$this->_deposit->setTransferredStatus(true);
+			case 'harvest-error':
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED);
```

Comment:
Given the staging server won't wait us to acknowledge updates, it's needed to set all the expected states. On a next issue, these binary flags can be replaced by a single state.